ResearchKit/ActiveTasks/ORKLocationRecorder.m

```diff
@@ -164,16 +198,59 @@ - (BOOL)isRecording {
     return [CLLocationManager locationServicesEnabled] && (self.locationManager != nil) && ([CLLocationManager authorizationStatus] > kCLAuthorizationStatusDenied);
 }
 
-- (void)reset {
-    [super reset];
-    
-    _logger = nil;
-}
-
 - (NSString *)mimeType {
     return @"application/json";
 }
 
+
+#pragma mark - total distance tracking
+
+- (BOOL)isStandingStill {
+    return ((ORKLocationRecorderConfiguration *)self.configuration).isStandingStill;
+}
+
+- (BOOL)relativeDistanceOnly {
+    return ((ORKLocationRecorderConfiguration *)self.configuration).relativeDistanceOnly;
+}
+
+- (NSTimeInterval)timestamp {
+    __block NSTimeInterval timestamp = 0;
+    dispatch_sync(_processingQueue, ^{
+        NSTimeInterval timeInterval = [_locationData.lastObject.timestamp timeIntervalSinceDate:self.startDate];
+        NSTimeInterval uptimeDelta = (self.referenceUptime > 0) ? (self.uptime - self.referenceUptime) : 0;
+        timestamp = (uptimeDelta + timeInterval);
+    });
+    return timestamp;
+}
+
+- (CLLocationDistance)distanceTraveled {
+    return _totalDistance;
+}
+
+- (BOOL)isOutdoors {
```

Comment:
It's already not always valid. In my house, I get accuracy of "10" which is the same thing I get walking back and forth along the sidewalk (as long as I'm not *near* my house).